### DISCUSSION TITLE
Compilation issue with tool chain compiler

### DISCUSSION THREAD

**Discussion by rajnkamr** [protobuf-c/protobuf-c#646] [Q&A]
I am using gcc and g++ local tool chain compiler and getting below compilation issue.
Tool chain is already at 4.8.2 
Any suggestion 

/mnt/local/rajnish.kumar/connector/FLOW_EFC-16.2-dev/6500/../3rdparty/protobuf-c/protobuf-c/protobuf-c.c: In function âvoid protobuf_c_buffer_simple_append(ProtobufCBuffer_, size_t, const uint8_t_)â:
/mnt/local/rajnish.kumar/connector/FLOW_EFC-16.2-dev/6500/../3rdparty/protobuf-c/protobuf-c/protobuf-c.c:200:45: error: invalid conversion from âvoid_â to âuint8_t_ {aka unsigned char_}â [-fpermissive]
   new_data = do_alloc(allocator, new_alloced);
                                             ^
/mnt/local/rajnish.kumar/connector/FLOW_EFC-16.2-dev/6500/../3rdparty/protobuf-c/protobuf-c/protobuf-c.c: In function âsize_t protobuf_c_message_get_packed_size(const ProtobufCMessage_)â:
/mnt/local/rajnish.kumar/connector/FLOW_EFC-16.2-dev/6500/../3rdparty/protobuf-c/protobuf-c/protobuf-c.c:656:61: error: invalid conversion from âconst void_â to âconst uint32_t_ {aka const unsigned int*}â [-fpermissive]
     rv += oneof_field_get_packed_size(field, qmember, member);

**Comment by rajnkamr** [comment]
Looks like this was due to use of g++ compiler.
I used gcc compiler and stuck up with following error

/usr/bin/ld: i386:x86-64 architecture of input file `protobuf-c/.libs/protobuf-c.o' is incompatible with i386 output
/usr/bin/ld: final link failed: Invalid operation

**Comment by rajnkamr** [comment]
It looks like the linker might be a creating problem, but after providing cross compiler linker it gave same error

Please suggest why it is trying to link /usr/bin/ld  instead of linker
LD=/mnt/sweng/tools/sysroots/6   5vse/v3/sysroots/centos_7.0/usr/x86_64-redhat-linux/bin/ld

  CCLD   protobuf-c/libprotobuf-c.la
/usr/bin/ld: i386:x86-64 architecture of input file `protobuf-c/.libs/protobuf-c.o' is incompatible with i386 output
/usr/bin/ld: final link failed: Invalid operation
collect2: ld returned 1 exit status
make[4]: **\* [protobuf-c/libprotobuf-c.la] Error 1
make[4]: **\* Waiting for unfinished jobs....

**Comment by lipnitsk** [comment]
OpenWrt builds this library for lots of platforms. Take a look [there](https://openwrt.org/packages/pkgdata/libprotobuf-c) for pointers.
